### PR TITLE
report: show disabled checkbox when all/no urls are third-party

### LINKED ISSUE #9096: Disable third party filter when no third parties present
In #8966 we remove the third party filter if all items are third party.

This creates this (unenjoyable) experience:

Run Lighthouse, see 6 issues. Check the third party filter box. Oh cool, I got 2 things to do.

Fix the 2 issues.

Run Lighthouse. See no checkbox and 4 items, but no indication (other than you parsing the URL) that they are third party.

Maybe we could style third party rows differently? What symbol would indicate "third party"?

### PR DESCRIPTION
**Pull request by connorjclark**
Fixes #9096

### CODE REVIEW

File: lighthouse-core/report/html/renderer/report-ui-features.js [hunk 1]
```diff
@@ -234,12 +234,11 @@ class ReportUIFeatures {
...
       const urlItems = this._getUrlItems(tableEl);
       const thirdPartyRows = this._getThirdPartyRows(tableEl, urlItems, this.json.finalUrl);
-      // If all or none of the rows are 3rd party, no checkbox!
-      if (thirdPartyRows.size === urlItems.length || !thirdPartyRows.size) return;
 
       // create input box
       const filterTemplate = this._dom.cloneTemplate('#tmpl-lh-3p-filter', this._templateContext);
-      const filterInput = this._dom.find('input', filterTemplate);
+      const filterInput =
+        /** @type {HTMLInputElement} */ (this._dom.find('input', filterTemplate));
```
Comment (patrickhulce) [inline lighthouse-core/report/html/renderer/report-ui-features.js]: man it'd really be nice to type this for some basic `HTMLXElement` inference

Comment (connorjclark) [reply lighthouse-core/report/html/renderer/report-ui-features.js]: but how often do we even select on just the tagname?

Comment (patrickhulce) [reply lighthouse-core/report/html/renderer/report-ui-features.js]: not enough to make me want to put in the effort to do it for real :)

image

**Comment by connorjclark** [comment]
image

**Comment by brendankenny** [comment]
I mean, looks good to me...maybe we'll issue a last call, land, and then fix later if folks have an issue with it :)

@paulirish @yuinchien

**Comment by yuinchien** [comment]
@brendankenny @paulirish 

Can we update the "Show 3rd Party Resources" button to the new design similar to the Performance Metrics toggle button?

Screen Shot 2019-07-10 at 12 08 00 PM

Screen Shot 2019-07-10 at 12 05 30 PM

The product logos for Wordpress should be aligned to the top next to the text description, I might be wrong but the latest screenshot appears to have the logo center aligned horizontally.

Screen Shot 2019-07-10 at 12 06 12 PM

**Comment by brendankenny** [comment]
> Can we update the "Show 3rd Party Resources" button to the new design similar to the Performance Metrics toggle button?

maybe in a follow up PR and limit this one to just keeping the option visible at all times?

If we change to the toggle that makes it easy to ship this PR without further style bikeshedding, cause we aren't going to keep it anyways :)

**Pull request merged**